cockroach-metrics/src/lib.rs

```diff
@@ -0,0 +1,1072 @@
+# This is a comment
+# TYPE some_metric counter
+# Another comment
+"#;
+        let metrics = PrometheusMetrics::parse(sample_metrics).unwrap();
+        assert!(metrics.metrics.is_empty());
+    }
+
+    #[test]
+    fn test_prometheus_histogram_parsing() {
+        let sample_metrics = r#"
+# TYPE sql_exec_latency histogram
+sql_exec_latency_bucket{le="0.001"} 10
+sql_exec_latency_bucket{le="0.01"} 25
+sql_exec_latency_bucket{le="0.1"} 100
+sql_exec_latency_bucket{le="1.0"} 200
+sql_exec_latency_bucket{le="+Inf"} 205
+sql_exec_latency_count 205
+sql_exec_latency_sum 45.2
+"#;
+
+        let metrics = PrometheusMetrics::parse(sample_metrics).unwrap();
+
+        // Check that histogram buckets are properly grouped under the base metric name
+        if let Some(MetricValue::Histogram(histogram)) =
+            metrics.metrics.get("sql_exec_latency")
+        {
+            assert_eq!(histogram.len(), 5, "Should have 5 histogram buckets");
+
+            // Buckets should already be sorted by le value
+            assert_eq!(histogram[0].le, 0.001);
+            assert_eq!(histogram[0].count, 10);
+            assert_eq!(histogram[1].le, 0.01);
+            assert_eq!(histogram[1].count, 25);
+            assert_eq!(histogram[4].le, f64::INFINITY);
+            assert_eq!(histogram[4].count, 205);
+
+            // Verify buckets are cumulative (and sorted)
+            for i in 1..histogram.len() {
+                assert!(
+                    histogram[i].count >= histogram[i - 1].count,
+                    "Histogram buckets should be cumulative"
+                );
+                assert!(
+                    histogram[i].le >= histogram[i - 1].le,
+                    "Histogram buckets should be sorted by le value"
+                );
+            }
+        } else {
+            panic!("sql_exec_latency histogram not found");
+        }
+
+        // Verify count and sum are stored as separate regular metrics
+        // (These are accessed by name since they're not in our CockroachMetric enum)
+        assert_eq!(
+            metrics.metrics.get("sql_exec_latency_count"),
+            Some(&MetricValue::Unsigned(205))
+        );
+        assert_eq!(
+            metrics.metrics.get("sql_exec_latency_sum"),
+            Some(&MetricValue::Float(45.2))
+        );
+
+        // Test strongly-typed histogram access
+        if let Some(histogram) =
+            metrics.get_metric_histogram(CockroachMetric::SqlExecLatency)
+        {
+            assert_eq!(histogram.len(), 5);
+        } else {
+            panic!(
+                "sql_exec_latency histogram not accessible via CockroachMetric enum"
+            );
+        }
+    }
+
+    #[test]
+    fn test_typed_cockroach_metrics() {
+        let sample_metrics = r#"
+leases_error 5
+sql_exec_latency_bucket{le="0.001"} 10
+sql_exec_latency_bucket{le="0.01"} 25
+sql_exec_latency_bucket{le="+Inf"} 30
+sql_exec_latency_count 30
+sql_exec_latency_sum 2.5
+"#;
```

Comment:
...and here (sorry):
```suggestion
        let sample_metrics = "\n\
            leases_error 5\n\
            sql_exec_latency_bucket{le="0.001"} 10\n\
            sql_exec_latency_bucket{le="0.01"} 25\n\
            sql_exec_latency_bucket{le="+Inf"} 30\n\
            sql_exec_latency_count 30\n\
            sql_exec_latency_sum 2.5\n\
        ";
```